### PR TITLE
Improve Kafka Avro support

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Fixes https://github.com/wso2-extensions/esb-connector-kafka/issues/61
Fixes https://github.com/wso2-extensions/esb-connector-kafka/issues/60
Fixes https://github.com/wso2-extensions/esb-connector-kafka/issues/59

## Goals
- Use default values defined in the schema when a field is missing
- Reject a wrong field value during Avro schema validation
- Set error properties to message context

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java [hunk 1]
```diff
@@ -395,19 +409,30 @@ private Object handleComplexAvroTypes(Object jsonString, Schema schema) {
         if (jsonString == null) {
             return null;
         }
+        JsonNode defaultValueNode = schema.getJsonProp("default");
+        boolean usingDefault = false;
```
Comment (Vathsan) [inline src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java:413]: It seems the Enum type supports default value https://avro.apache.org/docs/current/spec.html. Shall we check if we have handled that too?

File: src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java [hunk 2]
```diff
@@ -131,14 +137,26 @@ public void connect(MessageContext messageContext) {
...
+     * Sets error to context and handle.
+     *
+     * @param msgCtx      Message Context to set info
+     * @param e           Exception associated
+     * @param error       Error code
+     * @param errorDetail Error detail
+     */
+    private void handleError(MessageContext msgCtx, Exception e, Error error, String errorDetail) {
+
+        Utils.setError(msgCtx, e, error);
```
Comment (Vathsan) [inline src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java:156]: Shall we use setErrorPropertiesToMessage and remove the setError method? WDYT?

File: src/main/java/org/wso2/carbon/connector/utils/Utils.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+        setErrorPropertiesToMessage(msgCtx, e, error);
+    }
+
+    /**
+     * Sets the error code and error detail in message
+     *
+     * @param messageContext Message Context
+     * @param error          Error to be set
+     */
```
Comment (Vathsan) [inline src/main/java/org/wso2/carbon/connector/utils/Utils.java:55]: Lets update the java doc for exception parameter

Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/utils/Utils.java:44]: Fixed with ba1167b

File: src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java (hunk 2 above)
Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java]: Fixed with ba1167b

File: src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java (hunk 1 above)
Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java:413]: Fixed with ba1167b

**Review by Vathsan — APPROVED** [review]

**Pull request merged**